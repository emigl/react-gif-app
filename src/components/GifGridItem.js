import React from 'react'

const GifGridItem = ({id, title, url}) => {
    
    
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

export default GifGridItem
