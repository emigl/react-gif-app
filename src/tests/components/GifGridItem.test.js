import React from 'react'
import {shallow} from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Test de la card del gifGridItem', () => {

    const title = 'Esto es un titulo'
    const url = 'https://localhost:3000/algo.jpg'
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test('debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot()

    })

    test('debe de tener un parrafo con el title', () => {

        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })

    test('debe de tener la imagen igual al url y al alt de los props', () => {

        const img = wrapper.find('img')
        
        expect(img.props().src).toBe(url)
        // Se puede utilizar props().<nombre-atributo> o prop('nombre-atributo')
        expect(img.prop('alt')).toBe(title)
    })

    test('debe de tener la clase animate__fadeIn', () => {
        
        const div = wrapper.find('div')

        expect(div.prop('className').includes('animate__fadeIn')).toBe(true)

    })
    
})
