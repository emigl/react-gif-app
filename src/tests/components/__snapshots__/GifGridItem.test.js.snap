// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de la card del gifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Esto es un titulo"
    src="https://localhost:3000/algo.jpg"
  />
  <p>
    Esto es un titulo
  </p>
</div>
`;
